backend/__tests__/unit/controllers/userAuth.js

```diff
@@ -0,0 +1,116 @@
+  const connection = mongoose.connection;
+  connection.once('open', () => {
+      // connection successful
+  })
+});
+
+describe("Login", () => {
+    test("Successful Password Encryption", () => {
+        var password = "password"
+        expect(encrypt(password)).toBe("5baa61e4c9b93f3f0682250b6cf8331b7ee68fd8");
+    });
+
+
+    test("Failed Login - Incorrect Username", () => {
+      jest.setTimeout(30000);
+
+      const req = mockRequest({ body: {
+        username : "Fakename",
+        password : "test"
+      } });
+      var res = mockResponse({ hostname: 'tester',
+        status : function(statusCode) {
+          this.status = statusCode
+        },
+        json : function(body) {
+          this.json = body
+        },
+      });
+
+      var expectedResponse = {
+        "status": "200",
+        "success": "false",
+        "msg": "Username and password do not match"
+      }
+
+      return login(req, res).then(data => {
+        expect(data.json.status).toBe(expectedResponse.status);
+        expect(data.json.success).toBe(expectedResponse.success);
+        expect(data.json.msg).toBe(expectedResponse.msg);
+      });
+
+    });
+
+    test("Failed Login - Incorrect Password", () => {
+      jest.setTimeout(30000);
+
+      const req = mockRequest({ body: {
+        username : "Mitul",
+        password : "fakepass"
+      } });
+      var res = mockResponse({ hostname: 'tester',
+        status : function(statusCode) {
+          this.status = statusCode
+        },
+        json : function(body) {
+          this.json = body
+        },
+      });
+
+      var expectedResponse = {
+        "status": "200",
+        "success": "false",
+        "msg": "Username and password do not match"
+      }
+
+      return login(req, res).then(data => {
+        expect(data.json.status).toBe(expectedResponse.status);
+        expect(data.json.success).toBe(expectedResponse.success);
+        expect(data.json.msg).toBe(expectedResponse.msg);
+      });
+
+    });
+
+    // test("Successful Login - Correct Credentials", () => {
+    //   jest.setTimeout(30000);
+    //
+    //   const req = mockRequest({ body: {
+    //     username : "Mitul",
+    //     password : "test"
+    //   } });
+    //   var res = mockResponse({ hostname: 'tester',
+    //     status : function(statusCode) {
+    //       this.status = statusCode
+    //     },
+    //     json : function(body) {
+    //       this.json = body
+    //     },
+    //   });
+    //
+    //   var expectedResponse = {
+    //     "status": "200",
+    //     "success": "true",
+    //     "msg": "Login successful"
+    //   }
+    //
+    //   return login(req, res).then(data => {
+    //     expect(data.json.status).toBe(expectedResponse.status);
+    //     expect(data.json.success).toBe(expectedResponse.success);
+    //     expect(data.json.msg).toBe(expectedResponse.msg);
+    //   });
+    //
+    // });
```

Comment:
Is this supposed to be commented out? Or will this test be implemented in the future?

Reply:
Should be removed as we have done successful login code coverage in the integration test